docs/DeveloperGuide.md

```diff
@@ -85,6 +85,74 @@ This section describes some noteworthy details on how certain features are imple
 
 ### **Task Manager** <a name="task-manager"></a>
 
+TrackIt@NUS allows users to keep track of his/her tasks. The task manager is one of TrackIt@NUS's basic operations
+.
+ 
+ #### Rationale
+ 
+ The task manager is a crucial feature in TrackIt@NUS. Tasks are an integral part of any student's day-to-day life
+ . Hence, TrackIt@NUS includes a task manager for students to keep track of all their tasks. To better
+  support NUS students, a task can either belong to a module or not. When adding a task, users
+  can choose to the include the `m/MODULE_CODE` parameter in order to add a task that belongs to a module. When users
+   click into a specific module tab, they can see the tasks belonging to each module.
+   
+![ModuleTasks](images/ModuleTasks.png)
+    
+:information_source: A task does not have to belong a module. In this case, the module parameter of the task is
+ simply treated as null.
+     
+:warning: The module must exist (i.e. there must be a module with the specified `MODULE_CODE`) otherwise the add
+ command will not work.
+ 
+:bulb: To remove a task from a module, simply type `T edit INDEX m/` (use the `m/` prefix but leave the `MODULE_CODE
+` parameter empty).
+
+#### Current Implementation
+
+In this section, we will outline the key commands of the Task Manager, namely:
+* `AddTaskCommand`
+* `DeleteTaskCommand`
+* `EditTaskCommand`
+
+We will also elaborate on one more key function, namely `getModuleTasks`.
+
+Generally, this is how the Task Manager handles a command.
+ 
+ ![Add Task](images/AddTaskActivityDiagram.png)
+ 
+The add, delete, and edit commands are all implemented in similar ways. When they are executed they will:
+ * call on the relevant Model methods
+ * update the `UniqueTaskList` depending on the command
+ * Save the updated task list to `data/trackIter.json`
+ * return the relevant CommandResult message
+ 
+ The following shows the sequence diagram of the `AddTaskCommand`.
```

Comment:
Hmm do you think the name of objects should be longer? like model & task instead of m & t